From 92334c933e8dd9d2df67ef78a3ad39dff7f08077 Mon Sep 17 00:00:00 2001
From: Tobias Schramm <t.schramm@manjaro.org>
Date: Thu, 13 May 2021 15:39:09 +0200
Subject: [PATCH 04/16] ARM: dts: sun8i: v3s: add analog codec and frontend to
 V3s dts

The Allwinner V3s and V3 SoCs feature an integrated analog audio codec.
Additionally both have an analog frontend with mixers and amplifiers.
The analog frontend is identical to that on the Allwinner H3 SoC.
This commit adds both, the analog codec and its frontend to the V3s dtsi.

Signed-off-by: Tobias Schramm <t.schramm@manjaro.org>
---
 arch/arm/boot/dts/sun8i-v3s.dtsi | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-v3s.dtsi b/arch/arm/boot/dts/sun8i-v3s.dtsi
index 81319b939cea..f5a046b9e364 100644
--- a/arch/arm/boot/dts/sun8i-v3s.dtsi
+++ b/arch/arm/boot/dts/sun8i-v3s.dtsi
@@ -428,6 +428,25 @@ lradc: lradc@1c22800 {
 			status = "disabled";
 		};
 
+		codec: codec@1c22c00 {
+		        #sound-dai-cells = <0>;
+		        compatible = "allwinner,sun8i-v3s-codec";
+		        reg = <0x01c22c00 0x400>;
+		        interrupts = <GIC_SPI 29 IRQ_TYPE_LEVEL_HIGH>;
+		        clocks = <&ccu CLK_BUS_CODEC>, <&ccu CLK_AC_DIG>;
+		        clock-names = "apb", "codec";
+		        resets = <&ccu RST_BUS_CODEC>;
+		        dmas = <&dma 15>, <&dma 15>;
+		        dma-names = "rx", "tx";
+		        allwinner,codec-analog-controls = <&codec_analog>;
+			status = "disabled";
+		};
+
+		codec_analog: codec-analog@1c23000 {
+			compatible = "allwinner,sun8i-h3-codec-analog";
+			reg = <0x01c23000 0x4>;
+		};
+
 		uart0: serial@1c28000 {
 			compatible = "snps,dw-apb-uart";
 			reg = <0x01c28000 0x400>;
-- 
2.30.1

